### PR TITLE
Fix for partition function

### PR DESCRIPTION
**Pull request by pzharrington**
# PhysicsNeMo Pull Request

## Description
Fixes a bug related to `distribute_module` PyTorch<=2.10 that silently flips `requires_grad` on frozen parameters. Any non-sharded parameters should fall back to `Replicate` placement, but wrapping them in `nn.Parameter` should preserve `requires_grad`. This should not be needed in PyTorch>2.10 but  might as well include the fix here for robustness and clarity.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

**Review by jleinonen — APPROVED** [review]

File: physicsnemo/nn/module/embedding_layers.py [hunk 1]
```diff
@@ -127,7 +127,16 @@ def __init__(
         freqs = torch.arange(start=0, end=self.freq_embed_dim // 2, dtype=torch.float32)
         freqs = freqs / (self.freq_embed_dim // 2 - (1 if self.endpoint else 0))
         freqs = (1 / self.max_positions) ** freqs
-        self.register_buffer("freqs", freqs, persistent=False)
+        self.register_buffer("freqs", freqs)
+        self.register_load_state_dict_pre_hook(self._fill_missing_freqs)
+
+    @staticmethod
+    def _fill_missing_freqs(module, state_dict, prefix, *_args, **_kwargs):
+        """Backward compat: old checkpoints saved freqs as non-persistent.
```
Comment (CharlelieLrt) [inline physicsnemo/nn/module/embedding_layers.py:135]: Just a FYI, we have both type of checkpoints I think:
- some with the buffer
- some without the buffer
It's because iirc the `persistent=False` was added only < 1 year ago, so checkpoints from before that had the buffer.

But anyways, your solution works for both I think, so approving

**Review by CharlelieLrt — APPROVED** [review]
LGTM!

File: physicsnemo/nn/module/embedding_layers.py (hunk 1 above)
Comment (pzharrington) [reply physicsnemo/nn/module/embedding_layers.py:135]: Actually the change came here: https://github.com/NVIDIA/physicsnemo/pull/1343

Previously the `freqs` were stored as a plain tensor (not in the state dict), so when they were refactored into a buffer we used `persistent=False` to avoid having to do a state dict hook like in this PR. Unfortunately we still need that now to support some checkpoints trained by NVR 😭
